R/plots_reports.R

```diff
@@ -173,8 +173,8 @@ interactive_plot <- function(x, selected_spectrum, x2 = NULL, selected_spectrum2
   spectra_plot <- plot_OpenSpecy(x, x2 = x2, selected_spectrum = selected_spectrum, selected_spectrum2 = selected_spectrum2)
 
   # Extract intensity and wavenumber for the selected spectrum
-  selected_spectrum_points <- x$metadata %>% filter(row_number() == selected_spectrum)
-  selected_spectrum_intensity <- selected_spectrum_points$intensity
+  selected_spectrum_points <- x$metadata[selected_spectrum, ]
+  selected_spectrum_intensity <- x$spectra[, ..selected_spectrum]
```

Comment:
I had to change those lines to avoid an error. Does the selection still works as intended?

Reply:
Yeah should work the same, that is basically the logic for the filter_spec function.